src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -285,34 +375,29 @@ bool InputStateMachineEngine::ActionEscDispatch(const wchar_t wch,
 //      that can include many parameters.
 // Arguments:
 // - wch - Character to dispatch.
-// - cIntermediate - Number of "Intermediate" characters found - such as '!', '?'
-// - wchIntermediate - Intermediate character in the sequence, if there was one.
-// - rgusParams - set of numeric parameters collected while pasring the sequence.
-// - cParams - number of parameters found.
+// - intermediate - Intermediate character in the sequence, if there was one.
+// - parameters - set of numeric parameters collected while pasring the sequence.
 // Return Value:
 // - true iff we successfully dispatched the sequence.
 bool InputStateMachineEngine::ActionCsiDispatch(const wchar_t wch,
-                                                const unsigned short /*cIntermediate*/,
-                                                const wchar_t /*wchIntermediate*/,
-                                                _In_reads_(cParams) const unsigned short* const rgusParams,
-                                                const unsigned short cParams)
+                                                const std::optional<wchar_t> /*intermediate*/,
+                                                const std::basic_string_view<size_t> parameters)
 {
-    DWORD dwModifierState = 0;
+    DWORD modifierState = 0;
     short vkey = 0;
-    unsigned int uiFunction = 0;
-    unsigned int col = 0;
-    unsigned int row = 0;
+    unsigned int function = 0;
+    size_t col = 0;
+    size_t row = 0;
 
     // This is all the args after the first arg, and the count of args not including the first one.
-    const unsigned short* const rgusRemainingArgs = (cParams > 1) ? rgusParams + 1 : rgusParams;
-    const unsigned short cRemainingArgs = (cParams >= 1) ? cParams - 1 : 0;
+    const auto remainingArgs = parameters.size() > 1 ? parameters.substr(1) : parameters;

```

Comment:
I think there's one more of these patterns somewhere that I replaced and it needs to be fixed as well as this one.

Reply:
ugh look at us reimplementing `car` and `cdr`

Reply:
Found it and fixed it.